ui/src/dashboards/apis/index.ts

```diff
@@ -1,102 +1,161 @@
+
+  return {
+    ...dashboard,
+    labels: dashboard.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(dashboard.cells, dashboard.id),
   }
 }
 
-export const deleteDashboard = async dashboard => {
-  try {
-    return await AJAX({
-      method: 'DELETE',
-      url: dashboard.links.self,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
+export const createDashboard = async (
+  props: CreateDashboardRequest
+): Promise<Dashboard> => {
+  const dashboard = await client.dashboards.create(props)
+
+  return {
+    ...dashboard,
+    labels: dashboard.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(dashboard.cells, dashboard.id),
   }
 }
 
-export const addDashboardCell = async (dashboard, cell) => {
-  try {
-    return await AJAX({
-      method: 'POST',
-      url: dashboard.links.cells,
-      data: cell,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
+export const deleteDashboard = async (dashboard: Dashboard): Promise<void> => {
+  await client.dashboards.delete(dashboard.id)
+}
+
+export const updateDashboard = async (
+  dashboard: Dashboard
+): Promise<Dashboard> => {
+  const updated = await client.dashboards.update(dashboard.id, dashboard)
+
+  return {
+    ...updated,
+    labels: updated.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(updated.cells, updated.id),
   }
 }
 
-export const deleteDashboardCell = async cell => {
-  try {
-    return await AJAX({
-      method: 'DELETE',
-      url: cell.links.self,
+export const addCell = async (
+  dashboardID: string,
+  cell: NewCell
+): Promise<Cell> => {
+  const result = await client.dashboards.createCell(dashboardID, cell)
+
+  const cellWithID = {...result, dashboardID}
+
+  return cellWithID
+}
+
+export const updateCells = async (
+  id: string,
+  cells: Cell[]
+): Promise<Cell[]> => {
+  const result = await client.dashboards.updateAllCells(id, cells)
+
+  return addDashboardIDToCells(result, id)
+}
+
+export const deleteCell = async (
+  dashboardID: string,
+  cell: Cell
+): Promise<void> => {
+  await client.dashboards.deleteCell(dashboardID, cell.id)
+}
+
+export const addDashboardLabels = async (
+  dashboardID: string,
+  labels: ILabel[]
+): Promise<ILabel[]> => {
+  const addedLabels = await Promise.all(
+    labels.map(async label => {
+      return client.dashboards.addLabel(dashboardID, label.id)
     })
-  } catch (error) {
-    console.error(error)
-    throw error
-  }
+  )
+
+  return addedLabels as ILabel[]
 }
 
-export const editTemplateVariables = async templateVariable => {
-  try {
-    return await AJAX({
-      method: 'PUT',
-      url: templateVariable.links.self,
-      data: templateVariable,
+export const removeDashboardLabels = async (
+  dashboardID: string,
+  labels: ILabel[]
+): Promise<void> => {
+  await Promise.all(
+    labels.map(label => {
+      return client.dashboards.removeLabel(dashboardID, label.id)
```

Comment:
i believe the same for this one, `removeLabels`, (if its not implemented i think thats an oversight and could be added)

Reply:
will do in next PR